lib/src/api/engine/any/mod.rs

```diff
@@ -1,87 +1,131 @@
 //! Dynamic support for any engine
 //!
-//! # Examples
+//! SurrealDB supports various ways of storing and accessing your data. For storing data we support a number of
+//! key value stores. These are RocksDB, SpeeDB, TiKV, FoundationDB and an in-memory store. We call these
+//! local engines. RocksDB and SpeeDB are file-based, single node key value stores. TiKV and FoundationDB are
+//! are distributed stores that can scale horizontally across multiple nodes. The in-memory store does not persist
+//! your data, it only stores it in memory. All these can be embedded in your application, so you don't need to
+//! spin up a SurrealDB server first in order to use them. We also support spinning up a server externally and then
+//! access your database via WebSockets or HTTP. We call these remote engines.
 //!
-//! ```no_run
-//! use serde::{Serialize, Deserialize};
-//! use serde_json::json;
-//! use std::borrow::Cow;
-//! use surrealdb::sql;
-//! use surrealdb::engine::any::connect;
-//! use surrealdb::opt::auth::Root;
+//! The Rust SDK abstracts away the implementation details of the engines to make them work in a unified way.
+//! All these engines, whether they are local or remote, work exactly the same way using the same API. The only
+//! difference in the API is the endpoint you use to access the engine. Normally you provide the scheme of the engine
+//! you want to use as a type parameter to `Surreal::new`. This allows you detect, at compile, whether the engine
+//! you are trying to use is enabled. If not, your code won't compile. This is awesome but it strongly couples your
+//! application to the engine you are using. In order to change an engine you would need to update your code to
+//! the new scheme and endpoint you need to use and recompile it. This is where the any engine comes in.
 //!
-//! #[derive(Serialize, Deserialize)]
-//! struct Name {
-//!     first: Cow<'static, str>,
-//!     last: Cow<'static, str>,
-//! }
+//! The any engine allows you to use any engine as long as it was enabled when compiling. Unlike with the typed scheme,
```

Comment:
```suggestion
//! The `<any>`engine allows you to use any of the two engines as long as it was enabled when compiling. Unlike with the typed scheme,
```

@rushmorem I am unsure if there are only two engines but this way reading `<any>` as a type will stand out from the word "any"

Reply:
It's not just two engine @Ekwuno, by "use any engine" here I was referring to the fact that `Surreal<Any>` allows you to use any engine that SurrealDB supports. That's five engines currently, soon to be six.

Reply:
Ah! can we make it ` Surreal<Any>` then.. Was a bit confusing to read

Reply:
I can see how it's confusing but the name of the engine is actually `any` 🤣 `Surreal<Any>` is the type it creates but you are right, let me change that.

Reply:
Done!